File: src/main/java/org/mitallast/queue/crdt/rest/RestCrdtModule.java
package org.mitallast.queue.crdt.rest;

import com.google.inject.AbstractModule;

public class RestCrdtModule extends AbstractModule {
    @Override
    protected void configure() {
        bind(RestCrdtRouting.class).asEagerSingleton();
        bind(RestLWWRegister.class).asEagerSingleton();
        bind(RestGCounter.class).asEagerSingleton();
    }
}

File: src/main/java/org/mitallast/queue/crdt/rest/RestCrdtModule.kt
package org.mitallast.queue.crdt.rest

import com.google.inject.AbstractModule

class RestCrdtModule : AbstractModule() {
    override fun configure() {
        bind(RestCrdtRouting::class.java).asEagerSingleton()
        bind(RestLWWRegister::class.java).asEagerSingleton()
        bind(RestGCounter::class.java).asEagerSingleton()
        bind(RestGSet::class.java).asEagerSingleton()
    }
}
